week2/longest_substring.cc

```diff
@@ -0,0 +1,110 @@
+/* longest_substring.cc
+ * author: connie
+ *
+ * LONGEST SUBSTRING W/O REPEATING CHARACTERS
+ * URL:
+ * https://leetcode.com/problems/longest-substring-without-repeating-characters/
+ */
+
+// TODO: these includes don't seem to be working in new environment rn
+//       get rid of all the std:: at some point when I figure this out
+#include <iostream>
+#include <string>
+#include <algorithm>
+#include <cassert>
+
+//== DECLARATIONS
+int greatest_substr(std::string str);
+
+//== DEFINITIONS
+//#define DEBUG
+
+//== FUNCTIONS
+/*
+ * int longest_substr(string str)
+ * input: str - the string to evaluate
+ * output: int - size of the greatest substring with no repeated chars of str
+ * notes: this is the main implementation
+ */
+ int longest_substr(std::string str)
+ {
+   // null exception
+   if (str == "") return -1;
+   // optimize. don't iterate if size is already 1
+   if (str.size() == 1) return 1;
+
+   // we want to track our current greatest substring as we iterate
+   int start = 0; int end;
+   int max = 0;
+   // assume that the charset is limited size of ASCII charset
+   int charset[128] = {0};
+
+   // iterate thru the string accordingly to find the greatest substring
+   for (end = 0; end < str.size(); end++)
+   {
+#ifdef DEBUG
+     std::string greatest_substr = str.substr(start,end);
```

Comment:
oops yes, 2nd arg should be the strlen specifier. i had `end-start` there before, must have accidentally deleted it in the text editor by mistake